File: trunk/quests/global/items/CHRMRareLang.pl
sub EVENT_SCALE_CALC {
  # Used for charms that scale with number of rare languages learned
  my $langmastered = 0;
 
  #check each rare language: Old Erudian through Elder Dragon
  for (my $i = 11; $i <= 22; $i++) {
    if($client->GetLanguageSkill($i) == 100) {
      $langmastered++;
    }
  }
  $questitem->SetScale($langmastered/12);
}

File: global/items/CHRMRareLang.lua
-- CHRMRareLang

function event_scale_calc(e)

    local languages = 0;

    for i = 11,22,1 do
        if (e.owner:GetLanguageSkill(i) == 100) then
            languages = languages + 1;
        end
    end

    e.self:SetScale(languages/12);
end